### PR TITLE
Add WaitForStatus and WaitForVersion to resource package

### PR DESCRIPTION
**Pull request by rossf7**
I'm updating the Ingress Controller chart e2e tests (WIP PR https://github.com/giantswarm/kubernetes-nginx-ingress-controller/pull/26).

This includes using the `resource` package to install the chart. But I need to wait for the release to be installed. @MarcelMue you've already done this in chart-operator but I think its better to move here.

https://github.com/giantswarm/chart-operator/blob/4b7de9d1e73306f08e34a17b8504e03f1ac07126/integration/release/release.go#L15

I also extracted `WaitForVersion` for completeness.

### CODE REVIEW

File: pkg/framework/resource/resource.go [hunk 1]
```diff
@@ -93,3 +96,51 @@ func (r *Resource) InstallResource(name, values, channel string, conditions ...f
 
 	return nil
 }
+
+func (r *Resource) WaitForStatus(gsHelmClient *helmclient.Client, release string, status string) error {
```
Comment (MarcelMue) [inline pkg/framework/resource/resource.go]: HelmClient is part of the resource struct. Why pass it in again?

File: pkg/framework/resource/resource.go [hunk 2]
```diff
@@ -93,3 +96,51 @@ func (r *Resource) InstallResource(name, values, channel string, conditions ...f
...
+			return microerror.Mask(err)
+		}
+		if rc.Status != status {
+			return microerror.Maskf(releaseStatusNotMatchingError, "waiting for %q, current %q", status, rc.Status)
+		}
+		return nil
+	}
+
+	notify := func(err error, t time.Duration) {
+		log.Printf("getting release status %s: %v", t, err)
```
Comment (MarcelMue) [inline pkg/framework/resource/resource.go]: Do we use this different logging for better readability? I always find it hard to tell where logs are coming from.

File: pkg/framework/resource/resource.go [hunk 3]
```diff
@@ -93,3 +96,51 @@ func (r *Resource) InstallResource(name, values, channel string, conditions ...f
...
+
+	b := framework.NewExponentialBackoff(framework.ShortMaxWait, framework.LongMaxInterval)
+	err := backoff.RetryNotify(operation, b, notify)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	return nil
+}
+
+func (r *Resource) WaitForVersion(gsHelmClient *helmclient.Client, release string, version string) error {
```
Comment (MarcelMue) [inline pkg/framework/resource/resource.go]: Same here with helmClient.

File: pkg/framework/resource/resource.go (hunk 2 above)
Comment (xh3b4sd) [reply pkg/framework/resource/resource.go]: Lets please use the configured logger and the usual log structure.

**Review by rossf7 — COMMENTED** [review]
Damn yes I moved this in a bit of a hurry :( Should be good now. PTAL.

**Review by xh3b4sd — APPROVED** [review]

**Pull request merged**